prov/shm/src/smr_cq.c

```diff
@@ -53,19 +142,30 @@ int smr_cq_open(struct fid_domain *domain, struct fi_cq_attr *attr,
 		return -FI_ENOSYS;
 	}
 
-	util_cq = calloc(1, sizeof(*util_cq));
-	if (!util_cq)
+	smr_cq = calloc(1, sizeof(*smr_cq));
+	if (!smr_cq)
 		return -FI_ENOMEM;
 
-	ret = ofi_cq_init(&smr_prov, domain, attr, util_cq,
+	ret = ofi_cq_init(&smr_prov, domain, attr, &smr_cq->util_cq,
 			  &ofi_cq_progress, context);
 	if (ret)
 		goto free;
 
-	(*cq_fid) = &util_cq->cq_fid;
+	if (attr->flags & FI_PEER_CQ) {
+		smr_cq->peer_cq = ((struct fi_peer_cq_context *) context)->cq;
+	} else {
+		ret = smr_init_peer_cq(smr_cq);
+		if (ret)
+			goto cleanup;
+	}
+
+	smr_cq->util_cq.cq_fid.fid.ops = &smr_cq_fi_ops;
```

Comment:
should smr_cq->util_cq.cq_fid.ops be set to something other than the default util_cq_ops? It works fine now, however, as far as I recall we agreed to disable the rest of the operations on the CQ. 

As outlined in the man page: If manual progress is needed on the peer CQ, the owner should drive progress by using the fi_cq_read() function with the buf parameter set to NULL and count equal 0. The peer provider should set other functions that attempt to read the peer's CQ (i.e. fi_cq_readerr, fi_cq_sread, etc.) to return -FI_ENOSYS.

I understand why the code is setup this way, so both FI_PEER_CQ case and the standard CQ would work. However, I question if we should be disabling the rest of the callbacks in the FI_PEER_CQ case.

Reply:
Sorry for the delayed response. I was on vacation. Updated with some changes. Disabling the other CQ calls (except ofi_cq_read) makes sense to me. See if that looks better to you